doc/modules/compiler/utils.rst

```diff
@@ -181,16 +181,86 @@ functions with :ref:`mux-kernel <specifications/mux-compiler-spec:Function
 Attributes>` attributes as a source of truth to distinguish between kernels and
 other functions.
 
-HandleBarriersPass
-------------------
+WorkItemLoopsPass
+-----------------
+
+The ``WorkItemLoopsPass`` is responsible for adding explicit parallelism to
+implicitly parallel SIMT kernels. It does so by wrapping each kernel up in a
+triple-nested loop over all work-items in the work-group. Thus, kernels
+scheduled by this pass can be invoked once per work-group.
+
+The order in which work-items are executed is fairly flexible as per the
+programming models the oneAPI construction kit supports, but generally in
+ascending order from `0` to `N-1` through the innermost `X` dimension, followed
+by the `Y` dimension, and lastly the `Z` dimension.
+
+Conceptually, the pass transforms ``old_kernel`` into ``new_kernel`` in the
+example below:
+
+.. code:: cpp
+
+   void old_kernel(int *in, int *out) {
+     size_t id = get_global_id(0);
+     out[id] = in[id] * 4;
+   }
+
+   void new_kernel(int *in, int *out) {
+     for (size_t z = 0; z != get_local_size(2); z++) {
+       for (size_t y = 0, y != get_local_size(1); y++) {
+         for (size_t x = 0, x != get_local_size(0); x++) {
+           size_t id = get_global_id(0);
+           out[id] = in[id] * 4;
```

Comment:
it may be better to have an example that actually uses the z and y

Reply:
Yeah. `get_local_linear_id`  implicitly uses all three dimensions - would that work? Maybe then we could replace it with

``` c
(z * get_local_size(1) * get_local_size(0)) + (y * get_local_size(0)) + x.
```

Reply:
Done just that